packages/manager/apps/observability/src/components/infrastructures/region-selector/RegionSelector.component.tsx

```diff
@@ -23,35 +28,72 @@ export default function RegionSelector() {
     types: 'SHARED',
   });
 
+  const groupedInfrastructures = useMemo(() => {
+    if (!infrastructures) {
+      return {};
+    }
+    return groupByGeographicZone(infrastructures);
+  }, [infrastructures]);
+
+  const zones = useMemo(() => {
+    const keys = Object.keys(groupedInfrastructures);
+    // Put 'all' first, then other zones
+    return keys.includes('all') ? ['all', ...keys.filter((key) => key !== 'all')] : keys;
+  }, [groupedInfrastructures]);
+
+  const [activeZone, setActiveZone] = useState<string>('all');
+
+  useMemo(() => {
+    if (zones.length > 0 && !zones.includes(activeZone) && zones[0]) {
+      setActiveZone(zones[0]);
+    }
+  }, [zones, activeZone]);
```

Comment:
There should not be state update in useMemos.
`useMemo` is executed during renders, and there should not be state update during renders.

The most direct fix would be to use a `useEffect`, [or maybe no useEffect at all like this pattern](https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes).

But it is still weird.
What is that thing doing anyway ? (I can't get what it does)